parsec/core/fs/userfs/userfs.py

```diff
@@ -388,7 +394,14 @@ async def workspace_rename(self, workspace_id: EntryID, new_name: AnyEntryName)
                 updated_workspace_entry
             )
             await self.set_user_manifest(updated_user_manifest)
-            self.event_bus.send(CoreEvent.FS_ENTRY_UPDATED, id=self.user_manifest_id)
+            self.event_bus.send(
+                CoreEvent.FS_ENTRY_UPDATED,
+                id=self.user_manifest_id,
+                reason=FSEntryUpdatedReason.WORKSPACE_RENAME,
+                entry_id=updated_workspace_entry.id,
+                entry_source_name=EntryName(workspace_entry.name),
```

Comment:
`workspace_entry.name` is already an `EntryName`

I guess you got bitten by this incorrect typing:
https://github.com/Scille/parsec-cloud/blob/641558f9938fd27fc5008c3502c4e78915b01a1e/parsec/api/data/manifest.py#L78

Here we should have `name: EntryName` (according to what is defined in the `WorkspaceEntry.SCHEMA_CLS`).

So can you do fix `WorkspaceEntry.name` typing ;-)

Reply:
BTW I'm not sure about the `entry_source_name` here: unlike for FolderManifest we are not actually moving a workspace around but are just changing it name.
This is a subtle difference but the main idea is you can have multiple workspace with the same name (which is of course impossible within a folder !)

So I think we should use the `entry_old_name`/`entry_new_name` to mimic the name of the parameter of this function and to mark the difference with folder.

Reply:
aaaand same as above: we should call them `workspace_id`/`workspace_xxx_name`